src/Orchard.Web/Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -79,28 +79,33 @@ public class AutoroutePartHandler : ContentHandler {
         }
 
         private void ProcessAlias(AutoroutePart part) {
-            // Generate an alias if one as not already been entered.
-            if (String.IsNullOrWhiteSpace(part.DisplayAlias)) {
-                part.DisplayAlias = _autorouteService.Value.GenerateAlias(part);
-            }
+            if (!part.HasDraft())
```

Comment:
Note also that this does not prevent the creation of duplicate DisplayAlias values for concurrent creation of published or non-draftable content items

Reply:
I think that yes if a Content Item HasDraft even if there's a published version of the item when we save it as a draft then we should not need to ProcessAlias() since we will eventually do if we publish that draft later on. Also that content item would already have an Alias associated with it ; even if the title of the content item would have changed on that draft we should not alter the Alias of the currently published item. I think we should keep the draft url's as they are for now. This would be a separate PR and we would need to have a different route for them something like /draft/slug instead of /content/slug.